packages/grafana-ui/src/components/SingleStatShared/shared.ts

```diff
@@ -55,24 +55,34 @@ export const getSingleStatDisplayValues = (options: GetSingleStatDisplayValueOpt
 
   const values: DisplayValue[] = [];
 
-  for (const series of data) {
-    if (stat === 'name') {
-      values.push(display(series.name));
-    }
-
-    for (let i = 0; i < series.fields.length; i++) {
-      const column = series.fields[i];
-
-      // Show all fields that are not 'time'
-      if (column.type === FieldType.number) {
-        const stats = calculateStats({
-          series,
-          fieldIndex: i,
-          stats: [stat], // The stats to calculate
-          nullValueMode: NullValueMode.Null,
-        });
-        const displayValue = display(stats[stat]);
-        values.push(displayValue);
+  const calc = getStatsCalculators([stat]);
+  if (!calc || calc.length < 1) {
+    values.push({
+      numeric: 0,
+      text: 'Unknown Stat',
+    });
+  } else if (data) {
+    const isNumberStat = calc[0].resultType === FieldType.number;
+
+    for (const series of data) {
+      for (let i = 0; i < series.fields.length; i++) {
+        const field = series.fields[i];
+
+        if (isNumberStat && field.type !== FieldType.number) {
+          continue;
+        }
+
+        // Show all fields that are not 'time'
+        if (field.type !== FieldType.time) {
```

Comment:
This is usually fine... but we somehow need to migrate `last_time` existing stat.

That is a stat that says show `last` of any `time` column.